### PR TITLE
Add `TryFrom` for `ExternalPtr`

### PR DESCRIPTION
**Pull request by CGMossa**
We want to move away from `FromRobj`. On this journey, we have a few places, where there is hard-coded some logic, that is `from_robj`, etc.
This PR introduces the `#[extendr]` on `impl`-blocks using `use_try_from = true`. To that end, a lot of other wrapper-macros needs adjustment.

I'd appreciate thorough investigation of this, but I've spent months already figuring out if this works or not. And it should work.

### CODE REVIEW

File: extendr-api/src/error.rs [hunk 1]
```diff
@@ -161,6 +162,13 @@ impl std::fmt::Display for Error {
             Error::ExpectedExternalPtrType(_robj, type_name) => {
                 write!(f, "Incorrect external pointer type {}", type_name)
             }
+            Error::ExpectedExternalNonNullPtr(robj) => {
+                write!(
+                    f,
+                    "expected non-null pointer in externalptr, instead {:?}",
+                    robj
+                )
+            }
```
Comment (JosiahParry) [inline extendr-api/src/error.rs:171]: 🔥

**Comment by CGMossa** [comment]
This one needs testing. I made a test but forgot to include it..  I'm not including the full changes to `ExternalPtr` in this PR. I also discovered that `AsRef` / `AsMut` aren't applicable for us.

**Comment by JosiahParry** [comment]
I updated the `CHANGELOG.md` to reflect the changes here. Once the new checks pass, I believe this is ready for merging.

**Review by JosiahParry — APPROVED** [review]
Green lights! CHANGELOG has been updated to reflect addtional method additions. All tests and new integrations tests are clear!

**Pull request merged**